src/Uno.UI/UI/Xaml/Controls/ToggleSwitch/ToggleSwitch.mux.cs

```diff
@@ -0,0 +1,722 @@
+			}
+
+			if (_tpKnobBounds != null)
+			{
+				_knobBoundsSizeChanged.Disposable = null;
+			}
+
+			_tpCurtainBounds = null;
+			_tpCurtainClip = null;
+			_spCurtainTransform = null;
+			_tpKnob = null;
+			_tpKnobBounds = null;
+			_spKnobTransform = null;
+			_tpThumb = null;
+			_tpHeaderPresenter = null;
+
+			base.OnApplyTemplate();
+
+			var spCurtainIDependencyObject = GetTemplateChild("SwitchCurtain");
+			var spCurtainBoundsIDependencyObject = GetTemplateChild("SwitchCurtainBounds");
+			var spCurtainClipIDependencyObject = GetTemplateChild("SwitchCurtainClip");
+			var spKnobIDependencyObject = GetTemplateChild("SwitchKnob");
+			var spKnobBoundsIDependencyObject = GetTemplateChild("SwitchKnobBounds");
+			var spThumbIDependencyObject = GetTemplateChild("SwitchThumb");
+
+			_tpCurtainBounds = spCurtainBoundsIDependencyObject as FrameworkElement;
+			_tpCurtainClip = spCurtainClipIDependencyObject as UIElement;
+			_tpKnob = spKnobIDependencyObject as FrameworkElement;
+			_tpKnobBounds = spKnobBoundsIDependencyObject as FrameworkElement;
+			_tpThumb = spThumbIDependencyObject as Thumb;
+
+			var spThumbIUIElement = _tpThumb as UIElement;
+
+			var spCurtainIUIElement = spCurtainIDependencyObject as UIElement;
+			if (spCurtainIUIElement != null)
+			{
+				var spCurtainRenderTransform = spCurtainIUIElement.RenderTransform;
+				_spCurtainTransform = spCurtainRenderTransform as TranslateTransform;
+			}
+
+			var spKnobIUIElement = spKnobIDependencyObject as UIElement;
+			if (spKnobIUIElement != null)
+			{
+				Transform spKnobRenderTransform = spKnobIUIElement.RenderTransform;
+				_spKnobTransform = spKnobRenderTransform as TranslateTransform;
+			}
+
+			if (spThumbIUIElement != null && _tpThumb != null)
+			{
+				_tpThumb.DragStarted += DragStartedHandler;
+				_dragStarted.Disposable = Disposable.Create(() => _tpThumb.DragStarted -= DragStartedHandler);
+				_tpThumb.DragDelta += DragDeltaHandler;
+				_dragDelta.Disposable = Disposable.Create(() => _tpThumb.DragDelta -= DragDeltaHandler);
+				_tpThumb.DragCompleted += DragCompletedHandler;
+				_dragCompleted.Disposable = Disposable.Create(() => _tpThumb.DragCompleted -= DragCompletedHandler);
+				spThumbIUIElement.Tapped += TapHandler;
+				_tap.Disposable = Disposable.Create(() => spThumbIUIElement.Tapped -= TapHandler);
+			}
+
+			if (_tpKnob != null || _tpKnobBounds != null)
+			{
+				if (_tpKnob != null)
+				{
+					_tpKnob.SizeChanged += SizeChangedHandler;
+					_knobSizeChanged.Disposable = Disposable.Create(() => _tpKnob.SizeChanged -= SizeChangedHandler);
+				}
+
+				if (_tpKnobBounds != null)
+				{
+					_tpKnobBounds.SizeChanged += SizeChangedHandler;
+					_knobBoundsSizeChanged.Disposable = Disposable.Create(() => _tpKnobBounds.SizeChanged -= SizeChangedHandler);
+				}
+			}
+
+			UpdateHeaderPresenterVisibility();
+
+			UpdateVisualState(false);
+		}
+
+		private void PrepareState()
+		{
+			//base.PrepareState();
```

Comment:
Is this expected ? Should we add an empty `PrepareState` method in base class so if we add it later we won't forget to un-comment this?

Reply:
Yes, the `PrepareState` is internal virtual method on `DependencyObject`, but does nothing by default. I will add a comment here to ensure it is mentioned. When we add something similar, we will still need to remove call to this from the `ToggleSwitch` constructor anyway.